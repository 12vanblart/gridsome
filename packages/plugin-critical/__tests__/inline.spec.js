const path = require('path')
const { inlineCriticalCSS } = require('../lib/inline')

test('inline critical css', async () => {
  const filePath = path.join(__dirname, 'index.html')
  const html = await inlineCriticalCSS(filePath, 'a{color:red;}')

  expect(html).toMatchSnapshot()
})
