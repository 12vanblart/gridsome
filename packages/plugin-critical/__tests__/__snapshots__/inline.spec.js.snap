// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`inline critical css 1`] = `
"<!DOCTYPE html>
<html lang=\\"en\\">
  <head>
    <title>Gridsome</title>
    <meta charset=\\"utf-8\\"/>
    <meta name=\\"generator\\" content=\\"Gridsome v0.0.1\\"/>
    <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1, viewport-fit=cover\\"/>
    <style id=\\"___critical-css\\">a{color:red;}</style>
    <link rel=\\"preload\\" href=\\"/css/styles.css\\" as=\\"style\\" onload=\\"this.onload=null;this.rel='stylesheet'\\"><noscript><link rel=\\"stylesheet\\" href=\\"/css/styles.css\\"></noscript>
    <link rel=\\"preload\\" href=\\"/css/2.styles.css\\" as=\\"style\\" onload=\\"this.onload=null;this.rel='stylesheet'\\"><noscript><link rel=\\"stylesheet\\" href=\\"/css/2.styles.css\\"></noscript>
    <link rel=\\"prefetch\\" href=\\"/css/3.styles.css\\">
    <!-- removed blocking stylesheet -->
    <!-- removed blocking stylesheet -->
    <!-- removed blocking stylesheet -->
    <link rel=\\"preload\\" href=\\"/js/app.js\\" as=\\"script\\">
  </head>
  <body>
    <div id=\\"app\\"></div>
    <script src=\\"/js/app.js\\" defer></script>
  </body>
</html>
"
`;
